File: Data Structures & Algorithms/decode-ways/submission-0.java
class Solution {
    public int numDecodings(String s) {
        return dfs(s, 0);
    }

    int dfs(String s, int i) {
        if (i == s.length()) return 1;

        if (s.charAt(i) == '0') return 0;

        int res = dfs(s, i+1);

        if (i < s.length() - 1) {
            if (s.charAt(i) == '1' || (s.charAt(i) == '2' && s.charAt(i+1) < '7')) {
                res += dfs(s, i+2);
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        out_str=""
        for i in strs:
            out_str=out_str+str(len(i))+"#"+i
        return out_str

    def decode(self, s: str) -> List[str]:
        res=[]
        i=0
        j=0
        while(i<len(s)):
            l1=''
            while(s[j]!='#'):
                l1=l1+s[j]
                j=j+1
       
            n=int(l1)
            i=j+1
            res.append(s[i:i+n])
    
            i=i+n
    
            j=i
        return res    
